cql3/statements/alter_keyspace_statement.cc

```diff
@@ -84,10 +84,21 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
             auto new_ks = _attrs->as_ks_metadata_update(ks.metadata(), *qp.proxy().get_token_metadata_ptr(), qp.proxy().features());
 
             if (ks.get_replication_strategy().uses_tablets()) {
-                const std::map<sstring, sstring>& current_rfs = ks.metadata()->strategy_options();
-                for (const auto& [new_dc, new_rf] : _attrs->get_replication_options()) {
-                    auto it = current_rfs.find(new_dc);
-                    sstring old_rf = it != current_rfs.end() ? it->second : "0";
+                const std::map<sstring, sstring>& current_rf_per_dc = ks.metadata()->strategy_options();
+                auto new_rf_per_dc = _attrs->get_replication_options();
+                new_rf_per_dc.erase(ks_prop_defs::REPLICATION_STRATEGY_CLASS_KEY);
+                for (const auto& [new_dc, new_rf] : new_rf_per_dc) {
+                    sstring old_rf = "0";
+                    auto new_dc_in_current_mapping = current_rf_per_dc.find(new_dc);
+                    if (new_dc_in_current_mapping != current_rf_per_dc.end()) {
```

Comment:
nit: if `new_dc_in_current_mapping` is not intended to be used further down the road, you could use a short-living temporary variable, like:
```
if (auto it = current_rf_per_dc.find(new_dc); it != current_rf_per_dc.end()) {
    old_rf = it->second;
}
```

Reply:
Yes, thanks, I'll make this change. Perhaps it was needed later at some point, but the code changed and it's only needed within the `if`'s scope.

Reply:
Done.